c2rust-transpile/src/c_ast/mod.rs

```diff
@@ -256,6 +256,33 @@ impl TypedAstContext {
         }
     }
 
+    /// Predicate for structs that are used to implement C's `va_list`.
+    pub fn is_va_list(&self, typ: CTypeId) -> bool {
+        let resolved_ctype = self.resolve_type(typ);
+        match resolved_ctype.kind {
+            CTypeKind::Struct(record_id) => {
+                let r#struct = &self[record_id];
+                if let CDeclKind::Struct { name: Some(ref nam), .. } = r#struct.kind {
+                    return nam == "__va_list_tag"
+                } else {
+                    false
+                }
+            },
+            _ => false
+        }
+    }
+
+    /// Predicate for pointers to structs that are binary compatible with C's `va_list`.
+    pub fn is_pointer_to_va_list(&self, typ: CTypeId) -> bool {
```

Comment:
It's a bit more complex than this, depending on the architecture:

- On struct-architectures, e.g., x86_64, there's a structure called `struct __va_list_tag` and a `typedef struct __va_list_tag __builtin_va_list[1];` definition which decays to `struct __va_list_tag*` when passed to callees.

- On pointer-architectures like x86 and ARM, `__va_list_tag` doesn't exist, and the other one is just `typedef char* __builtin_va_list;` (or `void*` in some cases). I'm not sure if the type you'll see in the transpiler is `char*` or `__builtin_va_list`.